sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/events.scala

```diff
@@ -202,3 +203,62 @@ case class RenameFunctionEvent(
     name: String,
     newName: String)
   extends FunctionEvent
+
+trait PartitionEvent extends DatabaseEvent {
+  /**
+   * Name of the table that was touched.
+   */
+  val name: String
```

Comment:
`parts: Seq[CatalogTablePartition]` instead of `val name: String`? Table name is already inherited in `TableEvent`.

Reply:
Does it make sense to store `parts:  Seq[TablePartitionSpec]`? `CatalogTablePartition` contains `TablePartitionSpec` inside.

- Create and Alter partition have as parameter `Seq[CatalogTablePartition]`
- Rename and Drop partition have as parameter `Seq[TablePartitionSpec]`

Reply:
Got it. I was referring only a partial part of your code. In that case, we don't need anything here in `PartitionEvent`.